sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangeEpkImpl.java

```diff
@@ -3,125 +3,253 @@
-        return visitor.visit(this);
+    public static FeedRangeEpkImpl forFullRange() {
+        return fullRangeEPK;
     }
 
     @Override
-    public Mono<UnmodifiableList<Range<String>>> getEffectiveRanges(
-        final IRoutingMapProvider routingMapProvider,
-        final String containerRid,
-        final PartitionKeyDefinition partitionKeyDefinition) {
+    public Mono<Range<String>> getEffectiveRange(
+        IRoutingMapProvider routingMapProvider,
+        RxDocumentServiceRequest request,
+        Mono<Utils.ValueHolder<DocumentCollection>> collectionResolutionMono) {
 
-        return Mono.just(this.rangeList);
+        return Mono.just(this.range);
     }
 
     @Override
-    public Mono<UnmodifiableList<String>> getPartitionKeyRanges(
-        final IRoutingMapProvider routingMapProvider,
-        final String containerRid,
-        final PartitionKeyDefinition partitionKeyDefinition) {
-
-        return routingMapProvider
-            .tryGetOverlappingRangesAsync(
-                null,
-                containerRid,
-                this.range,
-                false,
-                null)
-            .flatMap(pkRangeHolder -> {
-                final ArrayList<String> rangeList = new ArrayList<>();
-
-                if (pkRangeHolder != null) {
-                    final List<PartitionKeyRange> pkRanges = pkRangeHolder.v;
-                    for (final PartitionKeyRange pkRange : pkRanges) {
-                        rangeList.add(pkRange.getId());
-                    }
+    public Mono<List<String>> getPartitionKeyRanges(
+        IRoutingMapProvider routingMapProvider,
+        RxDocumentServiceRequest request,
+        Mono<Utils.ValueHolder<DocumentCollection>> collectionResolutionMono) {
+
+        checkNotNull(
+            routingMapProvider,
+            "Argument 'routingMapProvider' must not be null");
+        checkNotNull(
+            request,
+            "Argument 'request' must not be null");
+        checkNotNull(
+            collectionResolutionMono,
+            "Argument 'collectionResolutionMono' must not be null");
+
+        MetadataDiagnosticsContext metadataDiagnosticsCtx =
+            BridgeInternal.getMetaDataDiagnosticContext(request.requestContext.cosmosDiagnostics);
+
+        return collectionResolutionMono
+            .flatMap(documentCollectionResourceResponse -> {
+
+                final DocumentCollection collection = documentCollectionResourceResponse.v;
+                if (collection == null) {
+                    throw new IllegalStateException("Collection cannot be null");
                 }
 
-                return Mono.just((UnmodifiableList<String>)UnmodifiableList.unmodifiableList(rangeList));
+                final String containerRid = collection.getResourceId();
+
+                return routingMapProvider
+                    .tryGetOverlappingRangesAsync(
+                        metadataDiagnosticsCtx,
+                        containerRid,
```

Comment:
What is the result if the container was deleted and recreated with the same name while the objects were already created? Would this return an exception?

Reply:
Had an issue here - because I was mixing Rid and Collection self-link. Changed that in the last iteration.